File: python/Lab11/Song.py
from Record import *
class Song(Record):
	def __init__(self,  record_genre,  artist_name,  subgenre_name,  producer_name,  record_name,  album_name,
                 release_year,  record_value,  record_length,  composition_year,  song_number,  composer_name):
				self.record_genre = record_genre
				self.artist_name = artist_name
				self.subgenre_name = subgenre_name
				self.producer_name = producer_name
				self.record_name = record_name
				self.album_name = album_name
				self.release_year = release_year
				self.record_value = record_value
				self.record_length = record_length
				self.composition_year = composition_year
				self.song_number = song_number
				self.composer_name = composer_name

File: java/lab3-9/work/src/main/java/ua/lviv/iot/Song.java
package ua.lviv.iot;

public class Song extends Record {

    private RecordGenre recordGenre;

    public String labelName;

    public Song(final String artistName, final RecordGenre recordGenre,
                final String recordName, final String albumName,
                final int recordLength, final String composerName,
                final String labelName) {
        super(artistName, recordGenre, recordName, albumName, recordLength, composerName);
        setLabelName(labelName);
    }

    public Song(final String artistName, final RecordGenre recordGenre,
                final String recordName, final String albumName,
                final int recordLength, final String composerName,
                final String labelName, final int id) {
        super(artistName, recordGenre, recordName, albumName, recordLength, composerName, id);
        setLabelName(labelName);
    }


    private void setLabelName(final String labelName) {
        this.labelName = labelName;
    }

    @Override
    public final String getHeaders() {
        return super.getHeaders() + ",labelName\n";
    }

    @Override
    public final String toCSV() {
        return super.getArtistName()  + ", " + super.toCSV() + ", " + labelName + "\n";
    }
}
